### ISSUE TITLE
Video and audio tracks does not start

### ISSUE THREAD

**Issue by aevaldas** [livekit/client-sdk-js#1026]
### Describe the bug

When using the Livekit demo (on iPhone 13), sometimes video does not start for local participant (and of course is not streamed to the remote participant).

![IMG_0139](https://github.com/livekit/client-sdk-js/assets/38099310/52c87093-fcd0-4033-a73f-47966b7736f1)

### Reproduction

1. Macbook: Join the room with cam & mic: https://meet.livekit.io/rooms/mob5-6jrt 
2. iPhone: Open link https://meet.livekit.io/rooms/mob5-6jrt
3. iPhone: Accept permissions
4. iPhone: Write the name
5. iPhone: Click "Join room"
6. iPhone: Check if camera feed is shown - here's the bug. If not, refresh the page.

7. iPhone: Accept the permissions.
8. iPhone: go back to step 5.

I have uploaded two videos, which shows the whole flow (including logs in video).

### Logs

```shell
Logs are in the [videos](https://drive.google.com/drive/folders/1-icuHdRj9SKC2yHS_HeVa-cV95mQPzEn?usp=share_link).
```

### System Info

```shell
iPhone 13, 
  iOS 17.4
    Safari
MacBook Pro M1, 14.3. 
  Safari 17.3 (19617.2.3).

Link: https://meet.livekit.io/rooms/mob5-6jrt
```

### Severity

blocking all usage of LiveKit

### Additional Information

We have the same issue on our platform, which uses Livekit SDK.

**Comment by lukasIO** [comment]
Thanks for the report. 
It looks like the second time the camera permissions re-acquiring the camera stream is blocked for some reason. I just tried with iOS 17.2 and cannot reproduce this behaviour. 
Will try also with 17.4 and see if there's any difference.

**Comment by aevaldas** [comment]
> It's hard to follow the logs in the video, could you post them in their entirety as text?

```
[Log] rawFacingMode – {rawFacingMode: "user"} (dba1bf7f-511427f33cbdd780.js, line 1)
[Log] awaiting getUserMedia promise (dba1bf7f-511427f33cbdd780.js, line 1, x2)
[Log] reacquiring camera track – {trackSid: undefined, trackSource: "camera", trackMuted: false, …} (dba1bf7f-511427f33cbdd780.js, line 1)
{trackSid: undefined, trackSource: "camera", trackMuted: false, trackEnabled: true, trackKind: "video"}Object
[Log] restarting track with constraints – {trackSid: undefined, trackSource: "camera", trackMuted: false, …} (dba1bf7f-511427f33cbdd780.js, line 1)
{trackSid: undefined, trackSource: "camera", trackMuted: false, trackEnabled: true, trackKind: "video", …}Object
[Log] video playback aborted, likely due to new play request (dba1bf7f-511427f33cbdd780.js, line 1)
[Log] re-acquired MediaStreamTrack – {trackSid: undefined, trackSource: "camera", trackMuted: false, …} (dba1bf7f-511427f33cbdd780.js, line 1)
{trackSid: undefined, trackSource: "camera", trackMuted: false, trackEnabled: true, trackKind: "video"}Object
[Log] track mute bounce got cancelled by an unmute event – {trackSid: undefined, trackSource: "camera", trackMuted: false, …}
{trackSid: undefined, trackSource: "camera", trackMuted: false, trackEnabled: true, trackKind: "video"}Object
[Log] setting video track unmuted – {trackSid: undefined, trackSource: "camera", trackMuted: false, …} (dba1bf7f-511427f33cbdd780.js, line 1)
{trackSid: undefined, trackSource: "camera", trackMuted: false, trackEnabled: true, trackKind: "video"}Object
[Log] track mute bounce got cancelled by an unmute event – {trackSid: undefined, trackSource: "camera", trackMuted: false, …}
{trackSid: undefined, trackSource: "camera", trackMuted: false, trackEnabled: true, trackKind: "video"}Object
[Log] {url: "wss://meet-uylq8icb.livekit.cloud"} ([name]-405583ddfedead02.js, line 1)
[Log] room event connectionStateChanged – {room: "", roomSid: "", identity: "", …} (dba1bf7f-511427f33cbdd780.js, line 1)
{room: "", roomSid: "", identity: "", event: "connectionStateChanged", args: ["connecting"]}Object
[Log] connecting to wss://meet-uylq8icb.livekit.cloud/rtc?access_token=eyJhbGciOiJIUzI1NiJ9.eyJuYW1lIjoiRGhkaGRoIiwidmlkZW8iOnsicm9vbSI6Im1vYjUtNmpydCIsInJvb21Kb2luIjp0cnVlLCJjYW5QdWJsaXNoIjp0cnVlLCJjYW5QdWJsaXNoRGF0YSI6dHJ1ZSwiY2FuU3Vic2NyaWJlIjp0cnVlfSwiaXNzIjoiQVBJVmVNQnBxd3lMa01OIiwiZXhwIjoxNzA3NDAwMTY4LCJuYmYiOjAsInN1YiI6IkRoZGhkaCJ9.npoyKyWHXwQVTiRAeeP2zBJoklZkVc7FpLQ4pYXvSxg&auto_subscribe=1&sdk=js&version=1.15.11&protocol=11&adaptive_stream=1 – {room: undefined, roomSid: undefined, identity: undefined} (dba1bf7f-511427f33cbdd780.js, line 1)
[Warning] Could not resume audio context – {room: "", roomSid: "", identity: "", …} (dba1bf7f-511427f33cbdd780.js, line 1)
{room: "", roomSid: "", identity: "", error: InvalidStateError: Failed to start the audio device}Object
[Log] room event audioPlaybackChanged – {room: "", roomSid: "", identity: "", …} (dba1bf7f-511427f33cbdd780.js, line 1)
{room: "", roomSid: "", identity: "", event: "audioPlaybackChanged", args: []}Object
[Log] {url: "wss://meet-uylq8icb.livekit.cloud"} ([name]-405583ddfedead02.js, line 1)
[Log] ping config – {room: undefined, roomSid: undefined, identity: undefined, …} (dba1bf7f-511427f33cbdd780.js, line 1)
{room: undefined, roomSid: undefined, identity: undefined, timeout: 15, interval: 5}Object
[Log] clearing ping interval – {room: undefined, roomSid: undefined, identity: undefined} (dba1bf7f-511427f33cbdd780.js, line 1)
[Log] start ping interval – {room: undefined, roomSid: undefined, identity: undefined} (dba1bf7f-511427f33cbdd780.js, line 1)
[Log] connected to Livekit Server edition: 1, version: 1.5.2, protocol: 13, region: Germany, nodeId: NM_DFRA1A_QG6MNjuLf5Qv, debugInfo:  – {room: "mob5-6jrt", roomSid: "RM_5YA67X9Muney", identity: "Dhdhdh"} (dba1bf7f-511427f33cbdd780.js, line 1)
[Log] room event signalConnected – {room: "mob5-6jrt", roomSid: "RM_5YA67X9Muney", identity: "Dhdhdh", …} (dba1bf7f-511427f33cbdd780.js, line 1)
{room: "mob5-6jrt", roomSid: "RM_5YA67X9Muney", identity: "Dhdhdh", event: "signalConnected", args: []}Object
[Log] setTrackEnabled – {room: "mob5-6jrt", roomSid: "RM_5YA67X9Muney", identity: "Dhdhdh", …} (dba1bf7f-511427f33cbdd780.js, line 1)
{room: "mob5-6jrt", roomSid: "RM_5YA67X9Muney", identity: "Dhdhdh", participantSid: "PA_bjRDKDUyfAEu", participantId: "Dhdhdh", …}Object
[Log] setTrackEnabled – {room: "mob5-6jrt", roomSid: "RM_5YA67X9Muney", identity: "Dhdhdh", …} (dba1bf7f-511427f33cbdd780.js, line 1)
{room: "mob5-6jrt", roomSid: "RM_5YA67X9Muney", identity: "Dhdhdh", participantSid: "PA_bjRDKDUyfAEu", participantId: "Dhdhdh", …}Object
[Log] setTrackEnabled – {room: "mob5-6jrt", roomSid: "RM_5YA67X9Muney", identity: "Dhdhdh", …} (dba1bf7f-511427f33cbdd780.js, line 1)
{room: "mob5-6jrt", roomSid: "RM_5YA67X9Muney", identity: "Dhdhdh", participantSid: "PA_bjRDKDUyfAEu", participantId: "Dhdhdh", …}Object
[Log] received server offer – {room: "mob5-6jrt", roomSid: "RM_5YA67X9Muney", identity: "Dhdhdh", …} (dba1bf7f-511427f33cbdd780.js, line 1)
{room: "mob5-6jrt", roomSid: "RM_5YA67X9Muney", identity: "Dhdhdh", RTCSdpType: "offer", sdp: "v=0\r↵o=- 4739762570015216362 1707399869 IN IP4 0.0…jlp\r↵a=ice-pwd:MXCLxNbCkLJynePBatqIjNzUHnDYlySK\r↵", …}Object
[Log] awaiting getUserMedia promise (dba1bf7f-511427f33cbdd780.js, line 1, x2)
[Info] publishing track – {room: "mob5-6jrt", roomSid: "RM_5YA67X9Muney", identity: "Dhdhdh", …} (dba1bf7f-511427f33cbdd780.js, line 1)
{room: "mob5-6jrt", roomSid: "RM_5YA67X9Muney", identity: "Dhdhdh", participantSid: "PA_bjRDKDUyfAEu", participantId: "Dhdhdh", …}Object
[Info] publishing track – {room: "mob5-6jrt", roomSid: "RM_5YA67X9Muney", identity: "Dhdhdh", …} (dba1bf7f-511427f33cbdd780.js, line 1)
{room: "mob5-6jrt", roomSid: "RM_5YA67X9Muney", identity: "Dhdhdh", participantSid: "PA_bjRDKDUyfAEu", participantId: "Dhdhdh", …}Object
[Warning] Could not resume audio context – {room: "mob5-6jrt", roomSid: "RM_5YA67X9Muney", identity: "Dhdhdh", …} (dba1bf7f-511427f33cbdd780.js, line 1)
{room: "mob5-6jrt", roomSid: "RM_5YA67X9Muney", identity: "Dhdhdh", error: InvalidStateError: Failed to start the audio device}Object
[Log] using video encoding – {maxBitrate: 1445000, maxFramerate: 30, priority: undefined} (dba1bf7f-511427f33cbdd780.js, line 1)
[Log] using svc with scalabilityMode L3T3_KEY (dba1bf7f-511427f33cbdd780.js, line 1)
[Log] encodings – [Object, {rid: "h", maxBitrate: 481666.6666666667, maxFramerate: 30}, {rid: "q", maxBitrate: 160555.55555555556, maxFramerate: 30}] (3) (dba1bf7f-511427f33cbdd780.js, line 1)
[Object, {rid: "h", maxBitrate: 481666.6666666667, maxFramerate: 30}, {rid: "q", maxBitrate: 160555.55555555556, maxFramerate: 30}]Array (3)
[Log] room event audioPlaybackChanged – {room: "mob5-6jrt", roomSid: "RM_5YA67X9Muney", identity: "Dhdhdh", …} (dba1bf7f-511427f33cbdd780.js, line 1)
{room: "mob5-6jrt", roomSid: "RM_5YA67X9Muney", identity: "Dhdhdh", event: "audioPlaybackChanged", args: [true]}Object
[Log] received trackPublishedResponse – {room: "mob5-6jrt", roomSid: "RM_5YA67X9Muney", identity: "Dhdhdh", …} (dba1bf7f-511427f33cbdd780.js, line 1)
{room: "mob5-6jrt", roomSid: "RM_5YA67X9Muney", identity: "Dhdhdh", cid: "8aeb3b33-001b-44d4-9811-467e85f35450", track: "TR_AMFgHQzoxR7kuT"}Object
[Log] update publication info – {trackSid: "TR_AMFgHQzoxR7kuT", trackName: "", track: undefined, …} (dba1bf7f-511427f33cbdd780.js, line 1)
{trackSid: "TR_AMFgHQzoxR7kuT", trackName: "", track: undefined, trackEnabled: true, trackEncrypted: false, …}Object
[Log] publishing audio with encodings – {room: "mob5-6jrt", roomSid: "RM_5YA67X9Muney", identity: "Dhdhdh", …} (dba1bf7f-511427f33cbdd780.js, line 1)
{room: "mob5-6jrt", roomSid: "RM_5YA67X9Muney", identity: "Dhdhdh", participantSid: "PA_bjRDKDUyfAEu", participantId: "Dhdhdh", …}Object
[Log] setting munged local description – {} (dba1bf7f-511427f33cbdd780.js, line 1)
[Log] sending answer – {room: "mob5-6jrt", roomSid: "RM_5YA67X9Muney", identity: "Dhdhdh", …} (dba1bf7f-511427f33cbdd780.js, line 1)
{room: "mob5-6jrt", roomSid: "RM_5YA67X9Muney", identity: "Dhdhdh", answerSdp: "v=0\r↵o=- 8040069723491493167 2 IN IP4 127.0.0.1\r↵s…:78:54\r↵a=ice-options:trickle\r↵a=sctp-port:5000\r↵"}Object
[Log] pc state change: from NEW to CONNECTING – {room: "mob5-6jrt", roomSid: "RM_5YA67X9Muney", identity: "Dhdhdh"} (dba1bf7f-511427f33cbdd780.js, line 1)
[Log] primary PC state changed 1 – {room: "mob5-6jrt", roomSid: "RM_5YA67X9Muney", identity: "Dhdhdh"} (dba1bf7f-511427f33cbdd780.js, line 1)
[Log] received trackPublishedResponse – {room: "mob5-6jrt", roomSid: "RM_5YA67X9Muney", identity: "Dhdhdh", …} (dba1bf7f-511427f33cbdd780.js, line 1)
{room: "mob5-6jrt", roomSid: "RM_5YA67X9Muney", identity: "Dhdhdh", cid: "ec58dfdf-8e49-463f-8f9f-6acea898ab74", track: "TR_VCADDE3rBdR5XA"}Object
[Log] update publication info – {trackSid: "TR_VCADDE3rBdR5XA", trackName: "", track: undefined, …} (dba1bf7f-511427f33cbdd780.js, line 1)
{trackSid: "TR_VCADDE3rBdR5XA", trackName: "", track: undefined, trackEnabled: true, trackEncrypted: false, …}Object
[Log] publishing video with encodings – {room: "mob5-6jrt", roomSid: "RM_5YA67X9Muney", identity: "Dhdhdh", …} (dba1bf7f-511427f33cbdd780.js, line 1)
{room: "mob5-6jrt", roomSid: "RM_5YA67X9Muney", identity: "Dhdhdh", participantSid: "PA_bjRDKDUyfAEu", participantId: "Dhdhdh", …}Object
[Log] get sender capabilities – {room: "mob5-6jrt", roomSid: "RM_5YA67X9Muney", identity: "Dhdhdh", …} (dba1bf7f-511427f33cbdd780.js, line 1)
{room: "mob5-6jrt", roomSid: "RM_5YA67X9Muney", identity: "Dhdhdh", cap: Object}Object
[Warning] silence detected on local audio track – {room: "mob5-6jrt", roomSid: "RM_5YA67X9Muney", identity: "Dhdhdh", …} (dba1bf7f-511427f33cbdd780.js, line 1)
{room: "mob5-6jrt", roomSid: "RM_5YA67X9Muney", identity: "Dhdhdh", participantSid: "PA_bjRDKDUyfAEu", participantId: "Dhdhdh", …}Object
[Log] starting to negotiate – {room: "mob5-6jrt", roomSid: "RM_5YA67X9Muney", identity: "Dhdhdh"} (dba1bf7f-511427f33cbdd780.js, line 1)
[Log] setting munged local description – {room: "mob5-6jrt", roomSid: "RM_5YA67X9Muney", identity: "Dhdhdh"} (dba1bf7f-511427f33cbdd780.js, line 1)
[Log] sending offer – {room: "mob5-6jrt", roomSid: "RM_5YA67X9Muney", identity: "Dhdhdh", …} (dba1bf7f-511427f33cbdd780.js, line 1)
{room: "mob5-6jrt", roomSid: "RM_5YA67X9Muney", identity: "Dhdhdh", offerSdp: "v=0\r↵o=- 9145098139705487063 2 IN IP4 127.0.0.1\r↵s…le\r↵a=sctp-port:5000\r↵a=max-message-size:262144\r↵"}Object
[Log] received server answer – {room: "mob5-6jrt", roomSid: "RM_5YA67X9Muney", identity: "Dhdhdh", …} (dba1bf7f-511427f33cbdd780.js, line 1)
{room: "mob5-6jrt", roomSid: "RM_5YA67X9Muney", identity: "Dhdhdh", RTCSdpType: "answer"}Object
[Log] setting munged remote description – {room: "mob5-6jrt", roomSid: "RM_5YA67X9Muney", identity: "Dhdhdh"} (dba1bf7f-511427f33cbdd780.js, line 1)
[Log] room event localTrackPublished – {room: "mob5-6jrt", roomSid: "RM_5YA67X9Muney", identity: "Dhdhdh", …} (dba1bf7f-511427f33cbdd780.js, line 1)
{room: "mob5-6jrt", roomSid: "RM_5YA67X9Muney", identity: "Dhdhdh", event: "localTrackPublished", args: Array}Object
[Log] room event localTrackPublished – {room: "mob5-6jrt", roomSid: "RM_5YA67X9Muney", identity: "Dhdhdh", …} (dba1bf7f-511427f33cbdd780.js, line 1)
{room: "mob5-6jrt", roomSid: "RM_5YA67X9Muney", identity: "Dhdhdh", event: "localTrackPublished", args: Array}Object
[Log] rawFacingMode – {rawFacingMode: "user"} (dba1bf7f-511427f33cbdd780.js, line 1)
[Log] on data channel 1, _reliable – {room: "mob5-6jrt", roomSid: "RM_5YA67X9Muney", identity: "Dhdhdh"} (dba1bf7f-511427f33cbdd780.js, line 1)
[Log] on data channel 3, _lossy – {room: "mob5-6jrt", roomSid: "RM_5YA67X9Muney", identity: "Dhdhdh"} (dba1bf7f-511427f33cbdd780.js, line 1)
[Log] video playback aborted, likely due to new play request (dba1bf7f-511427f33cbdd780.js, line 1)
[Log] pc state change: from CONNECTING to CONNECTED – {room: "mob5-6jrt", roomSid: "RM_5YA67X9Muney", identity: "Dhdhdh"} (dba1bf7f-511427f33cbdd780.js, line 1)
[Log] primary PC state changed 2 – {room: "mob5-6jrt", roomSid: "RM_5YA67X9Muney", identity: "Dhdhdh"} (dba1bf7f-511427f33cbdd780.js, line 1)
[Log] updating track subscription permissions – {room: "mob5-6jrt", roomSid: "RM_5YA67X9Muney", identity: "Dhdhdh", …} (dba1bf7f-511427f33cbdd780.js, line 1)
{room: "mob5-6jrt", roomSid: "RM_5YA67X9Muney", identity: "Dhdhdh", participantSid: "PA_bjRDKDUyfAEu", participantId: "Dhdhdh", …}Object
[Log] room event connectionStateChanged – {room: "mob5-6jrt", roomSid: "RM_5YA67X9Muney", identity: "Dhdhdh", …} (dba1bf7f-511427f33cbdd780.js, line 1)
{room: "mob5-6jrt", roomSid: "RM_5YA67X9Muney", identity: "Dhdhdh", event: "connectionStateChanged", args: ["connected"]}Object
[Log] room event connected – {room: "mob5-6jrt", roomSid: "RM_5YA67X9Muney", identity: "Dhdhdh", …} (dba1bf7f-511427f33cbdd780.js, line 1)
{room: "mob5-6jrt", roomSid: "RM_5YA67X9Muney", identity: "Dhdhdh", event: "connected", args: []}Object
[Log] rawFacingMode – {rawFacingMode: "user"} (dba1bf7f-511427f33cbdd780.js, line 1)
[Warning] silence detected on local audio track – {room: "mob5-6jrt", roomSid: "RM_5YA67X9Muney", identity: "Dhdhdh", …} (dba1bf7f-511427f33cbdd780.js, line 1)
{room: "mob5-6jrt", roomSid: "RM_5YA67X9Muney", identity: "Dhdhdh", participantSid: "PA_bjRDKDUyfAEu", participantId: "Dhdhdh", …}Object
[Log] room event localAudioSilenceDetected – {room: "mob5-6jrt", roomSid: "RM_5YA67X9Muney", identity: "Dhdhdh", …} (dba1bf7f-511427f33cbdd780.js, line 1)
{room: "mob5-6jrt", roomSid: "RM_5YA67X9Muney", identity: "Dhdhdh", event: "localAudioSilenceDetected", args: Array}Object
[Log] room event connectionQualityChanged – {room: "mob5-6jrt", roomSid: "RM_5YA67X9Muney", identity: "Dhdhdh", …} (dba1bf7f-511427f33cbdd780.js, line 1)
{room: "mob5-6jrt", roomSid: "RM_5YA67X9Muney", identity: "Dhdhdh", event: "connectionQualityChanged", args: Array}Object
[Log] upstream paused – {room: "mob5-6jrt", roomSid: "RM_5YA67X9Muney", identity: "Dhdhdh", …} (dba1bf7f-511427f33cbdd780.js, line 1)
{room: "mob5-6jrt", roomSid: "RM_5YA67X9Muney", identity: "Dhdhdh", participantSid: "PA_bjRDKDUyfAEu", participantId: "Dhdhdh", …}Object
[Log] upstream paused – {room: "mob5-6jrt", roomSid: "RM_5YA67X9Muney", identity: "Dhdhdh", …} (dba1bf7f-511427f33cbdd780.js, line 1)
{room: "mob5-6jrt", roomSid: "RM_5YA67X9Muney", identity: "Dhdhdh", participantSid: "PA_bjRDKDUyfAEu", participantId: "Dhdhdh", …}Object
```

**Comment by lukasIO** [comment]
I tried to reproduce this again and wasn't able to, are you still seeing this issue on meet.livekit.io?

**Comment by aevaldas** [comment]
> I tried to reproduce this again and wasn't able to, are you still seeing this issue on meet.livekit.io?

No, seems to be solved. Closing
